applications/StructuralMechanicsApplication/custom_elements/cable_element_3D2N.cpp

```diff
@@ -75,54 +75,16 @@ CableElement3D2N::CreateElementStiffnessMatrix(
   KRATOS_CATCH("")
 }
 
-void CableElement3D2N::CalculateLocalSystem(MatrixType &rLeftHandSideMatrix,
-                                            VectorType &rRightHandSideVector,
-                                            ProcessInfo &rCurrentProcessInfo) {
-
-  KRATOS_TRY
-  // calculate internal forces
-  BoundedVector<double, msLocalSize> internal_forces = ZeroVector(msLocalSize);
-  this->UpdateInternalForces(internal_forces);
-  // resizing the matrices + create memory for LHS
-
-  rLeftHandSideMatrix = ZeroMatrix(msLocalSize, msLocalSize);
-  // creating LHS
-  noalias(rLeftHandSideMatrix) =
-      this->CreateElementStiffnessMatrix(rCurrentProcessInfo);
-
-
-  if (this->mIsCompressed) {
-    rRightHandSideVector = ZeroVector(msLocalSize);
-  } else {
-    // create+compute RHS
-    rRightHandSideVector = ZeroVector(msLocalSize);
-    // update Residual
-    noalias(rRightHandSideVector) -= internal_forces;
-  }
-
-  // add bodyforces
-  if (this->HasSelfWeight()) noalias(rRightHandSideVector) += this->CalculateBodyForces();
-  KRATOS_CATCH("")
-}
-
 void CableElement3D2N::CalculateRightHandSide(
     VectorType &rRightHandSideVector, ProcessInfo &rCurrentProcessInfo) {
 
   KRATOS_TRY
   rRightHandSideVector = ZeroVector(msLocalSize);
 
-  if (!this->mIsCompressed) {
-    BoundedVector<double,msLocalSize> internal_forces =
-      this->GetConstitutiveLawTrialResponse(rCurrentProcessInfo,false);
-
-    BoundedMatrix<double, msLocalSize, msLocalSize> transformation_matrix =
-        ZeroMatrix(msLocalSize, msLocalSize);
-    this->CreateTransformationMatrix(transformation_matrix);
-
-
-    noalias(rRightHandSideVector) -= prod(transformation_matrix, internal_forces);
-  }
+  BoundedVector<double, msLocalSize> internal_forces = ZeroVector(msLocalSize);
+  this->UpdateInternalForces(internal_forces);
 
+  if (!this->mIsCompressed) noalias(rRightHandSideVector) -= internal_forces;

```

Comment:
separate issue: Does mIsCompressed have to be a member?

Cannot you compute it on the fly?